### PR TITLE
fix: Revise secret flags for Salesforce (#587)

### PR DESCRIPTION
**Pull request by zregvart**
This marks: clientId and userName as non-secret component properties.

### CODE REVIEW

File: connectors/salesforce-upsert-contact-connector/src/main/resources/camel-connector-schema.json [hunk 1]
```diff
@@ -18,10 +18,10 @@
...
-    "clientId": { "kind": "property", "displayName": "Client Id", "group": "security", "label": "common,security", "required": true, "type": "string", "javaType": "java.lang.String", "deprecated": false, "secret": true, "description": "OAuth Consumer Key of the connected app configured in the Salesforce instance setup. Typically a connected app needs to be configured but one can be provided by installing a package." },
+    "clientId": { "kind": "property", "displayName": "Client Id", "group": "security", "label": "common,security", "required": true, "type": "string", "javaType": "java.lang.String", "deprecated": false, "secret": false, "description": "OAuth Consumer Key of the connected app configured in the Salesforce instance setup. Typically a connected app needs to be configured but one can be provided by installing a package." },
```
Comment (jimmidyson) [inline connectors/salesforce-upsert-contact-connector/src/main/resources/camel-connector-schema.json:21]: Could you update commit message to indicate the clientId field is also now not secret (which I agree with btw)?

Comment (zregvart) [reply connectors/salesforce-upsert-contact-connector/src/main/resources/camel-connector-schema.json:21]: Updated to:

    fix: Revise secret flags for Salesforce (#587)

    This marks: clientId and userName as non-secret component properties.

**Review by jimmidyson — APPROVED** [review]

File: connectors/salesforce-upsert-contact-connector/src/main/resources/camel-connector-schema.json (hunk 1 above)
Comment (jimmidyson) [reply connectors/salesforce-upsert-contact-connector/src/main/resources/camel-connector-schema.json:21]: Thanks!

**Comment by rhuss** [comment]
@jimmidyson @zregvart It seems that editing the jsons alone is not good enough as although there are checked in, they are autogenerated from somewhere. I.e. when I do a 'mvn install' this changes gets overwritten again.

Any idea where this comes from ? (before I start digging ...)

**Comment by zregvart** [comment]
@rhuss it could be the combination of annotation processor (`camel/tooling/apt`) and camel-package-maven-plugin (`camel/tooling/maven/camel-package-maven-plugin`) that generate JSON files for Camel Catalog. This would mean that a new release of Camel would be needed see [CAMEL-11478](https://issues.apache.org/jira/browse/CAMEL-11478).

**Comment by rhuss** [comment]
Agreed. Should we make the generation of the json somewhat optional in order to decrease the chances that the old version is checked in by accident ?. E.g. we could introduc a profile 'schema-regen' and configure the plugin only for this profile.

**Pull request merged**